misc/images/bazel/Dockerfile

```diff
@@ -0,0 +1,30 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+MZFROM ubuntu-base
+
+ARG ARCH_GCC
+
+RUN apt-get update \
+    && TZ=UTC DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+        ca-certificates \
+        curl \
+        g++ \
+        gcc \
+    && groupadd --system --gid=999 materialize \
+    && useradd --system --gid=999 --uid=999 --create-home materialize
```

Comment:
I think the Dockerfile doesn't actually _use_ this user, does it? AFAICT there's no `USER materialize` so I think it just runs as root. Which is probably just fine? We could just let everything run as `root`, not bother creating the `materialize` user, and `COPY` without chowning. Alternatively we could add `USER materialize` to the end of this Dockerfile.

Reply:
Opted to remove the `materialize` user, I thought it was needed at some point, but I guess not!